### PR TITLE
Update new header for Alias, Index Templates, Policy Managed Indices, Indexes and Transform Jobs

### PR DESCRIPTION
**Pull request by tandonks**
### Description
New Header UI changes for Alias, Indexes, Templates, Transforms and Policy Managed Indexes for ISM dashboard pages

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

Policy Managed Indexes (NewUI)
Screenshot 2024-08-17 at 1 26 47 AM
Screenshot 2024-08-17 at 1 27 23 AM

Policy Managed Indexes (OldUI)
Screenshot 2024-08-17 at 1 28 10 AM
image

_____________________

Indexes (NewUI)
image
image
image

Indexes (OldUI)
image
image
image

_____________________

Index Aliases (NewUI)
image

Index Aliases (OldUI)
image

_____________________

Templates (NewUI)
image
image
image

Templates (OldUI)
image
image
image

___________________

Transforms (NewUI)
image
image
image
image

Transforms (OldUI)
image
image
image
image

### CODE REVIEW

File: public/pages/ChangePolicy/containers/ChangePolicy/ChangePolicy.tsx [hunk 1]
```diff
@@ -154,13 +160,25 @@ export class ChangePolicy extends Component<ChangePolicyProps, ChangePolicyState
       hasSubmitted,
     } = this.state;
 
-    return (
-      <div style={{ padding: "0px 25px" }}>
-        <EuiTitle size="l">
-          <h1>Change policy</h1>
-        </EuiTitle>
+    const uiSettings = getUISettings();
+    const useUpdatedUX = uiSettings.get("home:useNewHomePage");
```
Comment (astute-decipher) [inline public/pages/ChangePolicy/containers/ChangePolicy/ChangePolicy.tsx:164]: Remove this

File: public/pages/ChangePolicy/containers/ChangePolicy/ChangePolicy.tsx [hunk 2]
```diff
@@ -192,12 +210,12 @@ export class ChangePolicy extends Component<ChangePolicyProps, ChangePolicyState
 
         <EuiFlexGroup alignItems="center" justifyContent="flexEnd">
           <EuiFlexItem grow={false}>
-            <EuiButtonEmpty onClick={this.onCancel} data-test-subj="changePolicyCancelButton">
+            <EuiButtonEmpty size={useUpdatedUX ? "s" : "l"} onClick={this.onCancel} data-test-subj="changePolicyCancelButton">
```
Comment (astute-decipher) [inline public/pages/ChangePolicy/containers/ChangePolicy/ChangePolicy.tsx:213]: use undefined

File: public/pages/CreateIndex/containers/CreateIndex/CreateIndex.tsx [hunk 3]
```diff
@@ -45,7 +47,20 @@ export class CreateIndex extends Component<CreateIndexProps> {
   render() {
     const isEdit = this.isEdit;
 
-    return (
+    const uiSettings = getUISettings();
+    const useUpdatedUX = uiSettings.get("home:useNewHomePage");
```
Comment (astute-decipher) [inline public/pages/CreateIndex/containers/CreateIndex/CreateIndex.tsx:51]: remove this

File: public/pages/CreateIndexTemplate/containers/CreateIndexTemplate/CreateIndexTemplate.tsx [hunk 4]
```diff
@@ -64,7 +70,10 @@ class CreateIndexTemplate extends Component<CreateIndexTemplateProps> {
   };
 
   render() {
-    return (
+    const uiSettings = getUISettings();
+    const useUpdatedUX = uiSettings.get("home:useNewHomePage");
+
+    return !useUpdatedUX ? (
```
Comment (astute-decipher) [inline public/pages/CreateIndexTemplate/containers/CreateIndexTemplate/CreateIndexTemplate.tsx:76]: remove this

File: public/pages/CreateIndexTemplate/containers/TemplateDetail/TemplateDetail.tsx [hunk 5]
```diff
@@ -192,8 +196,79 @@ const TemplateDetail = (props: TemplateDetailProps, ref: Ref<FieldInstance>) =>
     </EuiFlexItem>
   );
 
-  return (
-    <>
+  const uiSettings = getUISettings();
```
Comment (astute-decipher) [inline public/pages/CreateIndexTemplate/containers/TemplateDetail/TemplateDetail.tsx:199]: Pass it as Props for TemplateDetails

File: public/pages/CreateIndexTemplate/containers/TemplateDetail/TemplateDetail.tsx [hunk 6]
```diff
@@ -217,51 +292,68 @@ const TemplateDetail = (props: TemplateDetailProps, ref: Ref<FieldInstance>) =>
...
       </EuiFlexGroup>
-      <EuiSpacer />
+    ) : (
+      <>
+        {!isEdit ? <HeaderControl controls={descriptionData} setMountPoint={setAppDescriptionControls} /> : null}
+        {isEdit ? (
+          <>
+            <HeaderControl controls={HeaderRight} setMountPoint={setAppRightControls} />
+          </>
+        ) : null}
```
Comment (astute-decipher) [inline public/pages/CreateIndexTemplate/containers/TemplateDetail/TemplateDetail.tsx:349]: Can be modified as. : 
```
{isEdit ? (
            <HeaderControl controls={HeaderRight} setMountPoint={setAppRightControls} />
        ) : <HeaderControl controls={descriptionData} setMountPoint={setAppDescriptionControls} />}
```

File: public/pages/Indices/components/IndexControls/IndexControls.tsx [hunk 7]
```diff
@@ -64,7 +77,44 @@ export default class IndexControls extends Component<IndexControlsProps, IndexCo
         ]
       : undefined;
 
-    return (
+    const uiSettings = getUISettings();
+    const useUpdatedUX = uiSettings.get("home:useNewHomePage");
+
+    return useUpdatedUX ? (
+      <EuiFlexGroup style={{ padding: "0px 5px 16px 5px" }} alignItems="center">
```
Comment (astute-decipher) [inline public/pages/Indices/components/IndexControls/IndexControls.tsx:84]: confirm this padding

File: public/pages/Templates/containers/TemplatesActions/index.tsx [hunk 8]
```diff
@@ -26,13 +27,18 @@ export default function TemplatesActions(props: TemplatesActionsProps) {
 
   const renderKey = useMemo(() => Date.now(), [selectedItems]);
 
+  const uiSettings = getUISettings();
+  const useUpdatedUX = uiSettings.get("home:useNewHomePage");
+
+  const size = useUpdatedUX ? "s" : "m";
```
Comment (astute-decipher) [inline public/pages/Templates/containers/TemplatesActions/index.tsx:33]: undefined

File: public/components/ContentPanel/ContentPanelActions.tsx [hunk 9]
```diff
@@ -17,15 +17,16 @@ interface ContentPanelActionsProps {
...
 
-const ContentPanelActions: React.SFC<ContentPanelActionsProps> = ({ actions }) => (
+const ContentPanelActions: React.SFC<ContentPanelActionsProps> = ({ actions, size }) => (
   <EuiFlexGroup justifyContent="spaceBetween" alignItems="center">
     {actions.map(({ text, buttonProps = {}, flexItemProps = {}, modal = null, children }, index) => {
       let button = children ? (
         children
       ) : (
-        <EuiButton {...buttonProps} data-test-subj={`${text}Button`}>
+        <EuiButton {...buttonProps} data-test-subj={`${text}Button`} size={size ? size : "m"}>
```
Comment (SuZhou-Joe) [inline public/components/ContentPanel/ContentPanelActions.tsx:29]: Just size is good enough, m is the default value of EuiButton size I recommber.

File: public/pages/ChangePolicy/components/ChangeManagedIndices/ChangeManagedIndices.tsx [hunk 10]
```diff
@@ -109,6 +109,7 @@ export default class ChangeManagedIndices extends Component<ChangeManagedIndices
               // @ts-ignore
               onChange={this.props.onChangeManagedIndices}
               onSearchChange={this.onManagedIndexSearchChange}
+              compressed={useUpdatedUX ? true : false}
```
Comment (SuZhou-Joe) [inline public/pages/ChangePolicy/components/ChangeManagedIndices/ChangeManagedIndices.tsx:112]:
```suggestion
              compressed={useUpdatedUX ? true : undefined}
```
undefined is preferred, and i just feel that it is totally OK to use a compressed ComboBox no matter the toggle is on or off.

**Review by SuZhou-Joe — APPROVED** [review]

**Comment by SuZhou-Joe** [comment]
@tandonks Please split the PR into several smaller PRs next time. It's a big pressure for maintainers to review such a huge PR.

**Pull request merged**